src/core/meshcop/joiner_router.cpp

```diff
@@ -141,14 +138,49 @@ ThreadError JoinerRouter::GetJoinerPort(uint16_t &aJoinerPort)
 
         if (tlv->GetType() == Tlv::kJoinerUdpPort)
         {
-            aJoinerPort = reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
-            ExitNow(error = kThreadError_None);
+            mJoinerUdpPort = reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
+            break;
         }
 
         cur += sizeof(Tlv) + tlv->GetLength();
     }
 
 exit:
+    return mJoinerUdpPort;
+}
+
+ThreadError JoinerRouter::SetJoinerUdpPort(uint16_t aJoinerUdpPort)
+{
+    ThreadError error = kThreadError_None;
+    uint8_t *data;
+    uint8_t *cur;
+    uint8_t *end;
+    uint8_t length;
+
+    mJoinerUdpPort = aJoinerUdpPort;
+
+    // update CommissioningData if it contains JoinerUdpPort TLV.
+    if ((data = mNetif.GetNetworkDataLeader().GetCommissioningData(length)) != NULL)
```

Comment:
I don't think the Joiner Router should be updating the Commissioner Dataset here since it may be propagated to other devices.  If the Joiner Router wants to override what's in the Commissioner Dataset, then it should do so without modifying the Commissioner Dataset.